pkg/providers/cloudstack/template.go

```diff
@@ -41,22 +29,18 @@ func generateTemplateBuilder(clusterSpec *cluster.Spec) (providers.TemplateBuild
 	return templateBuilder, nil
 }
 
-func getEtcdMachineSpec(clusterSpec v1alpha1.ClusterSpec, machineConfigs map[string]*v1alpha1.CloudStackMachineConfig) (*v1alpha1.CloudStackMachineConfigSpec, error) {
+func getEtcdMachineSpec(clusterSpec v1alpha1.ClusterSpec, machineConfigs map[string]*v1alpha1.CloudStackMachineConfig) *v1alpha1.CloudStackMachineConfigSpec {
 	var etcdMachineSpec *v1alpha1.CloudStackMachineConfigSpec
 	if clusterSpec.ExternalEtcdConfiguration != nil {
 		if clusterSpec.ExternalEtcdConfiguration.MachineGroupRef != nil && machineConfigs[clusterSpec.ExternalEtcdConfiguration.MachineGroupRef.Name] != nil {
 			etcdMachineSpec = &machineConfigs[clusterSpec.ExternalEtcdConfiguration.MachineGroupRef.Name].Spec
-		} else {
-			return etcdMachineSpec, errors.Errorf("getting MachineGroupRef")
 		}
-	} else {
-		return etcdMachineSpec, errors.Errorf("getting external etcd config")
```

Comment:
The cluster config is already being validated [here](https://github.com/aws/eks-anywhere/blob/c98621273ade48012679a66f9bcc62e72f40a49d/pkg/api/v1alpha1/cluster.go#L384), so there is no need for this